pandas/core/apply.py

```diff
@@ -1116,7 +1117,8 @@ def apply_empty_result(self) -> Series:
         )
 
     def apply_standard(self) -> DataFrame | Series:
-        f = self.f
+        # caller is responsible for ensuring that f is Callable
+        f = cast(Callable, self.f)
```

Comment:
This is the same approach as taken for the other cases (for example, [apply_str](https://github.com/pandas-dev/pandas/blob/66f08b951838b7be8624ed0b066dcbe34be07a79/pandas/core/apply.py#L532)).

I think we might be able avoid the casts altogether by passing `self.f` as an argument to the `apply_X` functions, if that would be a desirable change.

Reply:
I assume `self.f : Callable | None`? I think it would be nicer to `assert f is not None`

Reply:
`self.f` is typed as `AggFuncType` (defined [here](https://github.com/pandas-dev/pandas/blob/66f08b951838b7be8624ed0b066dcbe34be07a79/pandas/_typing.py#L160), which does not include `None`. Do we still have to check in that case?

Reply:
Sounds reasonable to me, might be good to check git blame and then cc the person who last touched this method.

Reply:
that would be @rhshadrach :slightly_smiling_face:

Reply:
Within the init of the Apply class, `self.f` is packed with args/kwargs when appropriate. This keeps that logic DRY; would passing f instead mean that logic needs to be repeated multiple places? If so, I'd be opposed to the change.

Another option would be to not pack `self.f` with args/kwargs at all, and only do so in the appropriate circumstances later in the code. That may be a better option altogether, but I'm not sure if there are other issues that might arise if one attempts this.

Reply:
I think delaying the arg packing of `self.f` might be an interesting option, since it looks like this is only used when `f` is `Callable` (so the `apply_default` case). But it seems like that might be a bigger change that affects more code, so I think it's better to explore that in a separate PR.